### ISSUE TITLE
How can I return a successful ready state to Ruby?

### ISSUE THREAD

**Issue by merefield** [meew0/discordrb#645]
Hey guys,

I've written a plugin for Discourse to leverage your great library, but have hit a challenge.

I want to return 'ready' state to the code I'm using to invoke the Bot.

When I call:

`bot = Discordrb::Commands::CommandBot.new token: SiteSetting.discord_bot_token, prefix: '!'` 

That gives me a bot 'object', but how can I determine that bot is ready?

```
    bot.ready do |event|
      puts "Logged in as #{bot.profile.username} (ID:#{bot.profile.id}) | #{bot.servers.size} servers"
    end
```
This will print out the bot is ready in console, but I need to run some Ruby in my core app (on a different thread) on the basis of the bot becoming ready.

I've tried to employ a Queue to communicate between threads, but bot.ready doesn't seem to be able to interact with it successfully (I can't see anything added to the Queue outside if I do so within bot.ready)

Here's an example of how i'm using the Queue:

![image](https://user-images.githubusercontent.com/35533304/61717721-849bac00-ad59-11e9-9cb9-d4d46c051502.png)

Basically I ultimately want to be able to start the bot with a button in the main app, and for a status to return to that app that the bot has launched successfully.

Stretch requirement:  how to shut that bot thread down programmatically?

**Comment by swarley** [comment]
To monitor when the bot has received `READY` use [#ready](https://www.rubydoc.info/github/meew0/discordrb/Discordrb%2FEventContainer:ready)
```ruby
bot.ready do
  # ... 
end
```


You can stop your bot with `Bot#stop`

**Comment by swarley** [comment]
In the future, if possible, questions regarding usage should be directed to our support channel `#ruby_discordrb` in the [community discord] (https://discord.gg/cyK3Hjm)

**Comment by merefield** [comment]
Closing issue, i'll follow up on the requested channel (I still have questions)
